packaging/package.sh

```diff
@@ -0,0 +1,243 @@
+IGNITE_VERSION=              # Main product version
+PACKAGING_DIR=               # Root directory of packaging script
+DEB_WORK_DIR=                # Main directory for building DEB packages
+RPM_WORK_DIR=                # Main directory for building RPM packages
+
+BUILD_DEB_FLAG=false         # Whether to build DEB
+BUILD_RPM_FLAG=false         # Whether to build RPM
+BUILD_FROM_SRC_FLAG=false    # Whether to build packages from sources
+
+
+
+###############
+#  FUNCTIONS  #
+###############
+
+# Usage help
+usage () {
+    cat <<EOF
+
+######################################################################
+#  Build RPM or DEB package from Apache Ignite's sources or binaries #
+######################################################################
+
+Prerequisites:
+     - RPM: binary archive with name 'apache-ignite-fabric-<version>-bin.zip'
+     - DEB: previously built corresponding RPM package
+
+Usage: ./$(basename ${BASH_SOURCE[0]}) --rpm,--deb [--src]
+
+Options:
+    --rpm, --deb     select package type for building (multiselect)
+
+    --src            prepare binary artifacts from source before packaging
+
+EOF
+}
+
+
+# Check and prepare build environment
+prepEnv () {
+    installCmd=""
+    packages="unzip curl alien gcc"
+    executables="rpmbuild unzip curl alien gcc"
+
+    # Check OS
+    name=$(cat /etc/*release | grep ^NAME | sed -r 's|.*"(.*)".*|\1|')
+    case ${name} in
+        "Ubuntu")
+            installCmd='apt --no-install-recommends'
+            packages="${packages} rpm"
+            ;;
+        "CentOS Linux")
+            installCmd="yum"
+            packages="${packages} rpm-build"
+            ;;
+        *)
+            echo "Unknown or unsupported linux detected"
+            echo "Will skip 'Check and prepare build environment' step"
+            echo "Please, prepare you environment manually"
+            ;;
+    esac
+
+    # Install missing software if necessary
+    installFlag=false
+    if [[ ! -z "${installCmd}" ]]; then
+        for executable in ${executables}; do
+            command -v ${executable} &>/dev/null || {
+                installFlag=true
+                break
+            }
+        done
+        if ${installFlag}; then
+            echo "[INFO] Software installation required root privileges"
+            sudo ${installCmd} install ${packages}
+        fi
+    fi
+}
+
+
+# Prepare binary artifacts from sources
+prepBin () {
+    echo "Packaging from sources is not implemented"
+}
+
+
+# Check that binary archive exists and try to download it from Apache Dist Archive is not
+getBin () {
+    IGNITE_VERSION=$(cat rpm/apache-ignite.spec | grep Version: | head -1 | sed -r 's|.*:\s+(.*)|\1|')
+    BIN_NAME="apache-ignite-fabric-${IGNITE_VERSION}-bin.zip"
+    binPreparedFlag=false
+
+    if [ -f "${BIN_NAME}" ]; then
+    binPreparedFlag=true
+    fi
+    if ! ${binPreparedFlag}; then
+        curl -O https://archive.apache.org/dist/ignite/${IGNITE_VERSION}/${BIN_NAME}
```

Comment:
how about cp $PACKAGING_DIR/../target/bin/apache-ignite-fabric-2.5.0-bin.zip . ?
Otherwise,
% cat apache-ignite-fabric-2.5.0-bin.zip                                                                                                                 14:41
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>404 Not Found</title>

Reply:
Fixed